### PR TITLE
Cleanup public api

### PR DESCRIPTION
**Pull request by Nutomic**
As noted by @MV-GH there are some parts of the public API which seem completely unnecessary:
- `PersonSortType` is only used internally, with automatic conversion from `SortType`, moved it and made it private
- Fields like `PostAggregate.hot_rank` or `PostAggregate.community_id` seem irrelevant for clients, marked them as `serde(skip)`

### CODE REVIEW

File: crates/db_schema/src/aggregates/structs.rs [hunk 1]
```diff
@@ -87,21 +90,32 @@ pub struct PostAggregates {
...
+  #[serde(skip)]
   pub community_id: CommunityId,
+  #[serde(skip)]
   pub creator_id: PersonId,
+  #[serde(skip)]
   pub controversy_rank: f64,
+  #[serde(skip)]
   pub instance_id: InstanceId,
   /// A rank that amplifies smaller communities
+  #[serde(skip)]
```
Comment (dessalines) [inline crates/db_schema/src/aggregates/structs.rs:118]: featured_local, featured_community are used by lemmy-ui. 

I grepped and it *seems* like the others are good / redundant with the postview.

Comment (MV-GH) [reply crates/db_schema/src/aggregates/structs.rs:118]: Both those you can get from `Post`

Comment (dessalines) [reply crates/db_schema/src/aggregates/structs.rs:118]: Ahh okay, thx.

**Review by dessalines — APPROVED** [review]

**Comment by cloudrac3r** [comment]
Could you add `newest_comment_time` back to this struct? I'm sad to be missing it.

Context: An instance updated to 0.19.1, and now that `newest_comment_time` is gone, my client can no longer show an indicator of which threads are currently active. I like being able to scroll through and jump into threads that happen to have active conversations without actually having to sort the whole app by active.

**Comment by dessalines** [comment]
@cloudrac3r Open up an issue for that if you would, its an easy fix.

**Pull request merged**